code/game/machinery/recycle_vendor.dm

```diff
@@ -205,15 +219,20 @@
 		for(var/toadd in matter)
 			combinedmats[toadd] += matter[toadd]
 		var/sellvalue = saleworthy_items[getthisone] * 0.8 // 20% fee
+		sellvalue = round(sellvalue) // don't make it round and you won't lose your money
 		var/datum/transaction/T = new(-sellvalue, "", "Recycling payout for [getthisone.name]", src)
 		T.apply_to(silo.my_account)
 		qdel(getthisone) // we first add the item to the garbage queue
 
-	saleworthy_items.Cut(saleworthy_items.Find(stufftorecycle[1]),saleworthy_items.Find(length(stufftorecycle)))
+	if(length(stufftorecycle) > 1)
+		saleworthy_items.Cut() // we melted it all, because we could afford it all.
+	else
+		saleworthy_items.Remove(stufftorecycle[1])
 	stufftorecycle = null // then we remove what should be the remaining references
 	playsound(loc, pick('sound/items/polaroid1.ogg', 'sound/items/polaroid2.ogg'), 50, 1)
 	silo.addmaterial(combinedmats)
 	spawn_money(combinedvalue, loc)
+	silo.updatesubsidy()
```

Comment:
in any case where the AME loses money and can go below The `MINIMUM_BUDGET` Value, it checks if funds are low enough to subsidize.